lib/screens/triptych_player/validator.ex

```diff
@@ -0,0 +1,47 @@
+defmodule Screens.TriptychPlayer.Validator do
+  @moduledoc false
+
+  @doc """
+  Determines whether a term is a valid player name -> screen ID mapping.
+  """
+  @spec validate(term()) :: :ok | {:error, reason :: String.t()}
+  def validate(mapping) do
+    with :ok <- validate_map(mapping),
+         :ok <- validate_string_keys(mapping) do
+      validate_screen_id_values(mapping)
+    end
+  end
+
+  defp validate_map(mapping) when is_map(mapping), do: :ok
+  defp validate_map(_), do: {:error, "Not a map"}
+
+  defp validate_string_keys(mapping) do
+    invalid_keys =
+      mapping
+      |> Map.keys()
+      |> Enum.reject(&is_binary/1)
+
+    case invalid_keys do
+      [] -> :ok
+      keys -> {:error, "Mapping contains non-string keys: #{inspect(keys)}"}
+    end
+  end
+
+  defp validate_screen_id_values(mapping) do
+    triptych_screen_ids = MapSet.new(Map.values(mapping))
+    all_screen_ids = MapSet.new(Screens.Config.State.screen_ids())
```

Comment:
Should this be filtered down to just triptychs so we only look for the difference of the mapping and `triptych_v2` screen IDs? Likely overkill, but could add some confidence that only triptych IDs are in the mapping.

Reply:
Yes, definitely. I added support for passing a filter function to `Config.State.screen_ids` and then didn't use it in my own new code. 🤦